src/di.jl

```diff
@@ -0,0 +1,195 @@
+for (ADGradient, fbackend) in ((:EnzymeADGradient, :AutoEnzyme),
+                               (:ZygoteADGradient, :AutoZygote))
+  @eval begin
+
+    struct $ADGradient{B, E} <: ADBackend
+      backend::B
+      extras::E
+    end
+
+    function $ADGradient(
+      nvar::Integer,
+      f,
+      ncon::Integer = 0,
+      c::Function = (args...) -> [];
+      x0::AbstractVector = rand(nvar),
+      kwargs...,
+    )
+      backend = $fbackend()
+      extras = DifferentiationInterface.prepare_gradient(f, backend, x0)
+      return $ADGradient(backend, extras)
+    end
+
+    function gradient(b::$ADGradient, f, x)
+      g = DifferentiationInterface.gradient(f, b.backend, x, b.extras)
+      # g = DifferentiationInterface.gradient(f, b.extras, b.backend, x)
```

Comment:
```suggestion
      # g = DifferentiationInterface.gradient(f, b.extras, b.backend, x)
```
?? Please remove dead code

Reply:
I forgot to add a comment, it's not dead code but a the code for the release 0.6.0 of DI.jl.
I don't would like to use it with the next major version of ADNLPModels.jl.

Reply:
Ok, but still let's not have commented lines like this. You can add this in an issue or a draft PR.